### ISSUE TITLE
strange connectivity error

### ISSUE THREAD

**Issue by whyrusleeping** [ipfs/go-ipfs#2561]
Trying to ping mars from my desktop (which has a prexisting connection to mars):

```
> ipfs ping QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ
PING QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ.
Ping error: session shutdown
```

The error appears to come from yamux. Will need to investigate further.

**Comment by whyrusleeping** [comment]
I think this is an issue with yamux, where the underlying connection breaks but yamux doesnt notice until we try to use it, resulting in this error.

**Comment by Kubuxu** [comment]
Was this fixed by #2689?

**Comment by Stebalien** [comment]
I believe this has been fixed.
